content/docs/user-guide/experiment-management/checkpoints.md

```diff
@@ -0,0 +1,454 @@
+              -p seed,lr,weight_decay --live dvclive python train.py
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`-c / --checkpoint` option of the `dvc stage add` command defines the checkpoint
+file or directory.
+
+The rest of the `dvc stage add` command sets up our dependencies for running the
+training code, which parameters we want to track (which are defined in the
+_params.yaml_), some configurations for our plots, showing the training metrics,
+and specifying where the logs produced by the training process will go.
+
+After running the command above to setup your _train_ stage, your _dvc.yaml_
+should have the following code.
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+  train:
+    cmd: python train.py
+    deps:
+      - data/MNIST
+      - train.py
+    params:
+      - seed
+      - lr
+      - weight_decay
+    outs:
+      - model.pt:
+          checkpoint: true
+    plots:
+      - predictions.json:
+          cache: false
+    live:
+      dvclive:
+        summary: true
+        html: true
+```
+
+Before we go any further, this is a great point to add these changes to your Git
+history. You can do that with the following commands:
+
+```bash
+git add .
+git commit -m "created DVC pipeline"
+```
+
+Now that you know how to enable checkpoints in a DVC pipeline, let's move on to
+setting up checkpoints in your code.
+
+## Registering checkpoints in your code
+
+Take a look at the _train.py_ file and you'll see how we train a convolutional
+neural network to classify handwritten digits. The main area of this code most
+relevant to checkpoints is when we iterate over the training epochs.
+
+This is where DVC will be able to track our metrics over time and where we will
+add our checkpoints to give us the points in time with our model that we can
+switch between.
+
+Now we need to enable checkpoints at the code level. We are interested in
+tracking the metrics along with each checkpoint, so we'll need to add a few
+lines of code.
+
+In the `train.py` file, import the `dvclive` package with the other imports:
+
+```bash
+import dvclive
+```
+
+Then update the following lines of code in the `main` method inside of the
+training epoch loop.
+
+```python
+# Iterate over training epochs.
+for i in range(1, EPOCHS+1):
+    # Train in batches.
+    train_loader = torch.utils.data.DataLoader(
+            dataset=list(zip(x_train, y_train)),
+            batch_size=512,
+            shuffle=True)
+    for x_batch, y_batch in train_loader:
+        train(model, x_batch, y_batch, params["lr"], params["weight_decay"])
+    torch.save(model.state_dict(), "model.pt")
+    # Evaluate and checkpoint.
+    metrics = evaluate(model, x_test, y_test)
+    for k, v in metrics.items():
+        print('Epoch %s: %s=%s'%(i, k, v))
+```
+
+The line `torch.save(model.state_dict(), "model.pt")` updates the checkpoint
+file.
```

Comment:
This isn't an update to the existing code, though, right? I'm not sure this line is needed, or maybe it belongs somewhere else.

Reply:
Right, it's not an update to the code. It's just an explanation of where checkpoints are created in the code. I added it per this comment https://github.com/iterative/dvc.org/pull/2373#discussion_r622475955